src/pages/Detail/Detail.js

```diff
@@ -1,9 +1,149 @@
+
+  componentDidMount = () => {
+    this.fetchData(this.restaurantsAddr);
+    this.fetchData(this.foodsAddr);
+    this.fetchData(this.imgAddr);
+    this.fetchReviewData(1, 5);
+  };
+
+  setRequestNumLimit = () => {
+    const { restaurants } = this.state;
+
+    restaurants.review_count.total % 10 === 0
+      ? (this.reviewRequestNumLimit = restaurants.review_count.total / 10)
+      : (this.reviewRequestNumLimit =
+          Math.floor(restaurants.review_count.total / 10) + 1);
+  };
+
+  handleReviewEdit = () => {
+    console.log('edit');
+  };
+
+  handleReviewDel = reviewId => {
+    fetch(
+      `http://${IP_ADDRESS}:8000/restaurants/${this.props.match.params.id}/review`,
+      {
+        method: 'DELETE',
+        body: JSON.stringify({
+          review_id: reviewId,
+        }),
+      }
+    ).then(() => this.reFetchReviewData(1, 5));
+  };
+
+  sortAddr = addr => {
+    const splitAddr = addr.split('/');
+
+    return parseInt(splitAddr[splitAddr.length - 1]) ==
+      this.props.match.params.id
+      ? splitAddr[0]
+      : splitAddr[splitAddr.length - 1];
+  };
+
+  fetchReviewData = (min, max) => {
+    fetch(
+      `http://${IP_ADDRESS}:8000/restaurants/${this.props.match.params.id}/review?limit=${this.reviewRequestNum}&rating-min=${min}&rating-max=${max}`,
+      {
+        method: 'GET',
+      }
+    )
+      .then(res => res.json())
+      .then(res => {
+        this.setState({ review: [...this.state.review, ...res.result] });
+        this.reviewRequestNum += 1;
+      });
+  };
+
+  reFetchReviewData = (min, max) => {
+    fetch(
+      `http://${IP_ADDRESS}:8000/restaurants/${this.props.match.params.id}/review?limit=1&rating-min=${min}&rating-max=${max}`,
+      {
+        method: 'GET',
+      }
+    )
+      .then(res => res.json())
+      .then(res => {
+        this.setState({ review: res.result });
+      });
+  };
+
+  fetchData = addr => {
+    fetch(`http://${IP_ADDRESS}:8000/${addr}`, {
+      method: 'GET',
+    })
+      .then(res => res.json())
+      .then(res => {
+        const name = this.sortAddr(addr);
+        console.log(name, res);
+        this.setState({
+          [name]: res.result,
+        });
+      });
+  };
+
   render() {
-    return <div>detail</div>;
+    const { restaurants, food, image, review } = this.state;
+    this.state.restaurants && this.setRequestNumLimit();
+    console.log(this.props);
+
+    return (
+      <section className="detailPage">
+        {image && <StoreImgBox imagesData={image} />}
+        <div className="detailMain">
+          {restaurants && food && (
+            <StoreInfo
+              fetchData={this.fetchData}
+              storeId={this.props.match.params.id}
+              restaurantsData={restaurants}
+              foodsData={food}
+            />
+          )}
+          {review.length !== 0 &&
+            restaurants &&
+            this.reviewRequestNumLimit !== undefined && (
```

Comment:
꼭 0과 undefined임을 강조할 의도가 아니었다면
```suggestion
          {!!review.length &&
            restaurants &&
            !!this.reviewRequestNumLimit && (
```
위와 같이 작성할 수도 있습니다